charts/newrelic-infrastructure-beta/templates/kubelet/daemonset.yaml

```
@@ -0,0 +1,275 @@
+            - name: "NRI_KUBERNETES_CLUSTERNAME"
+              value: {{ include "newrelic.cluster" . | required "cluster name must be set" }}
+            {{- if .Values.verboseLog }}
+            - name: NRI_KUBERNETES_VERBOSE
+              value: "1"
+            {{- end }}
+            - name: "NRI_KUBERNETES_NODENAME"
+              valueFrom:
+                fieldRef:
+                  apiVersion: "v1"
+                  fieldPath: "spec.nodeName"
+            # Required to connect to the kubelet
+            - name: "NRI_KUBERNETES_NODEIP"
+              valueFrom:
+                fieldRef:
+                  apiVersion: "v1"
+                  fieldPath: "status.hostIP"
+            {{- with .Values.kubelet.extraEnv }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+          {{- with .Values.kubelet.extraEnvFrom }}
+          envFrom: {{ toYaml . | nindent 12 }}
+          {{- end }}
+          volumeMounts:
+            - name: nri-kubernetes-config
+              mountPath: /etc/newrelic-infra/integrations.d/nri-kubernetes-config.yml
+              subPath: nri-kubernetes-config.yml
+            - mountPath: /tmp
+              name: tmpfs-tmp
+            {{- with .Values.kubelet.extraVolumeMounts }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+          {{- with .Values.kubelet.resources }}
+          resources: {{ toYaml . | nindent 12 }}
+          {{- end }}
+        - name: agent
+          image: "{{ .Values.images.agent.registry }}/{{ .Values.images.agent.repository }}:{{ .Values.images.agent.tag }}"
+          command: [ "/sbin/tini", "--", "/usr/bin/newrelic-infra" ]
+          imagePullPolicy: {{ .Values.images.agent.pullPolicy }}
+          securityContext:
+          {{- if .Values.privileged }}
+            privileged: true
+          {{- else }}
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          {{- end }}
+          ports:
+            - containerPort: 8003
+          env:
+            - name: "NRIA_OVERRIDE_HOSTNAME_SHORT"
+              valueFrom:
+                fieldRef:
+                  apiVersion: "v1"
+                  fieldPath: "spec.nodeName"
+            - name: "NRIA_FEATURES"
+              value: "docker_enabled:false"
+            - name: "NRIA_HTTP_SERVER_ENABLED"
+              value: "true"
+            - name: "NRIA_HTTP_SERVER_PORT"
+              value: "8003"
+            - name: NRIA_LICENSE_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "newrelic.licenseSecretName" . | required "licenseKey or customSecretName and customSecretLicenseKey must be set" }}
+                  key: {{ include "newrelic.licenseSecretKey" . | required "licenseKey or customSecretName and customSecretLicenseKey must be set" }}
+            {{- /* Ugly hack to check if .Values.enableProcessMetrics is defined: */}}
+            {{- if ne (print .Values.enableProcessMetrics) "<nil>" }}
```

We could use this slightly less ugly hack:
https://github.com/newrelic/helm-charts/pull/534/files#diff-c36211c7f3d9d611c1101c50f0dd080378972f7d7e815d89c56872474b1a240fR27

```suggestion
            {{- Check if enableProcessMetrics is present in the map */}}
            {{- if get .Values.enableProcessMetrics) | kindIs "bool" }}
```